tests/conftest.py

```diff
@@ -34,15 +32,35 @@ def xvfb():
 
 @pytest.yield_fixture(scope='function')
 def xterm_window(xvfb):
-    with NamedTemporaryFile() as tf:
-        proc = subprocess.Popen(
-            ['xterm', '-T', 'My window title',
-             '-e', 'echo "$WINDOWID" > "{}"; bash'.format(tf.name)])
-        time.sleep(.100)
-        with open(tf.name, 'rb') as fp:
-            window_id = int(fp.read())
-    yield XtermProcessInfo(proc=proc, window_id=window_id)
-    proc.terminate()
+    """
+    Create an xterm window test fixture.  This fixture opens a new xterm
+    window and yields its the X window id.  Upon test completion the xterm
+    process is cleaned up.
+    :param xvfb:
+    :return:
+    """
+    directory = tempfile.mkdtemp()
```

Comment:
It would probably be better to use the `tmpdir` fixture, so you don't need to take care of cleanup, etc..

https://pytest.org/latest/builtin.html?highlight=builtin%20fixtures#builtin-fixtures-function-arguments

i.e.:

```python
xterm_pipe_path = str(tmpdir.join('xterm_pipe'))
```